pkg/libzmix/bbs/bbs.go

```diff
@@ -0,0 +1,709 @@
+	var err C.ExternError
+
+	code := C.bbs_verify_context_set_signature(r.data, signature.Buffer, &err)
+	if code != 0 {
+		return errors.Errorf("error occurred setting signature to BBS verify context: %s", C.GoString(err.message))
+	}
+
+	return nil
+}
+
+//Finish verifies a BBS+ signature for a set of messages with a BBS public key
+func (r *VerifyContext) Finish() bool {
+	var err C.ExternError
+	code := C.bbs_verify_context_finish(r.data, &err)
+	return code == 0
+}
+
+type ProofContext Handle
+
+//ProofContextInit creates the proof asynchronous
+func ProofContextInit() (ProofContext, error) {
+	var err C.ExternError
+
+	handle := C.bbs_create_proof_context_init(&err)
+	if err.code != 0 {
+		return ProofContext{}, errors.Errorf("error occurred initializing BBS create proof context: %s", C.GoString(err.message))
+	}
+
+	return ProofContext{data: handle}, nil
+}
+
+type ProofMessageType int
+
+const (
+	Revealed ProofMessageType = 1
+	HiddenProofSpecificBlinding ProofMessageType = 2
+	HiddenExternalBlinding ProofMessageType = 3
+)
+
+//AddProofMessage adds a proof message to the proof context
+func (r *ProofContext) AddProofMessages(messages []string, messageType ProofMessageType, blindingFactor HandleByteBuffer) error {
+	var err C.ExternError
+	var proofMessageType C.proof_message_t
+
+	switch messageType {
+	case 1:
+		proofMessageType = C.Revealed
+	case 2:
+		proofMessageType = C.HiddenProofSpecificBlinding
+	case 3:
+		proofMessageType = C.HiddenExternalBlinding
+	}
+
+	for _, m := range messages {
+		var message C.ByteBuffer
+		message.len = (C.long)(len(m))
+		cm := C.CString(m)
+		message.data = (*C.uchar)(unsafe.Pointer(cm))
+		code := C.bbs_create_proof_context_add_proof_message_bytes(r.data, message, proofMessageType, blindingFactor.Buffer, &err)
+		if code != 0 {
+			return errors.Errorf("error occurred adding message bytes to BBS create proof context: %s", C.GoString(err.message))
+		}
+	}
+
+	return nil
+}
+
+//SetSignature sets the signature on the ProofContext
+func (r *ProofContext) SetSignature(signature *Signature) error {
+	var err C.ExternError
+
+	code := C.bbs_create_proof_context_set_signature(r.data, signature.Buffer, &err)
+	if code != 0 {
+		return errors.Errorf("error occurred setting signature to BBS create proof context: %s", C.GoString(err.message))
+	}
+
+	return nil
+}
+
+//SetPublicKey sets the bbs public key on the ProofContext
+func (r *ProofContext) SetPublicKey(publicKey HandleByteBuffer) error {
+	var err C.ExternError
+
+	code := C.bbs_create_proof_context_set_public_key(r.data, publicKey.Buffer, &err)
+	if code != 0 {
+		return errors.Errorf("error occurred setting secret key to BBS create proof context: %s", C.GoString(err.message))
+	}
+
+	return nil
+}
+
+//SetNonce sets the nonce on the ProofContext
+func (r *ProofContext) SetNonce(nonce []byte) error {
+	var err C.ExternError
+	var nonceBuffer C.ByteBuffer
+	defer C.bbs_byte_buffer_free(nonceBuffer)
+
```

Comment:
Since you aren't allocating the ByteBuffer in the lib, I don't think you need this `free` call.